lib/grandpa/message_handler.go

```diff
@@ -561,6 +561,10 @@ func (s *Service) VerifyBlockJustification(hash common.Hash, justification []byt
 		return err
 	}
 
+	if !hash.Equal(fj.Commit.Hash) {
+		return fmt.Errorf("%w: %s not equal %s", ErrJustificationMismatch, fj.Commit.Hash.Short(), hash.Short())
```

Comment:
mention which hash you expected and which was the received hash, since it's hard to know here which hash is which